### ISSUE TITLE
Explicit declaration of profiles

### ISSUE THREAD

**Issue by ViktorHaag** [IMSGlobal/caliper-central#215]
Caliper Events should declare the _governing profile_ that can help the receiver of an event interpret the contents of the Event. Here's a simple example:

- In the basic use of the `NavigationEvent`, the `object` property can be any `DigitalResource`.
- However, in the use of this event within the context of the _Survey profile_, perhaps `NavigationEvent.object` _must_ be a Questionnaire.

Which set of rules should the receiver of the navigation event use to interpret (validate) the event?

If we add an optional, top-level property on the event (`Event.profile`), this can carry a Caliper vocabulary term that would identify the profile to use to interpret the event's contents.

Note that this would be _different to the context_ of the Event; the context enumerates the vocabulary terms that are known within the Caliper vocabulary for the JSON-LD document you're reading, however, it doesn't contain many rules about what kinds of _values_ can be found within the fields inside the document, especially in cases where the use of a value in one profile is a _more specific use of a particular family of types_ (like requiring a `Person` instead of merely an `Agent`, or a `WebPage` instead of merely a `DigitalResource`).

**Comment by bracken** [comment]
Yes, I've heard of a couple people wanting this and the certification suite, I believe, has to guess the profile to be able to evaluate it correctly.

**Comment by ViktorHaag** [comment]
I think we should add this to core, and we should seek to add it for 1.2, so that it would be made available to all extensions after 1.2; otherwise, adding it will in extension to 1.2 will be a bit complicated and funky.

**Comment by bracken** [comment]
I think I have some concerns with this because if it's optional it won't be used a lot and so doesn't solve the problems we hope for, if it's required, I feel like it is maybe kind of limiting in some ways and people will just use GeneralProfile a lot. I'll try to think through my concerns to discuss in the call next week. 

I'm pretty sure I'm +1 on this proposal, but feel we should figure out what the negatives will be too, because there are some I think.

**Comment by arwhyte** [comment]
I assume that we would need to make `Event.profile<Term>` an optional (though recommended) property in order to assure backwards compatibility.  

I'd like to think that both producers and consumers would prefer to emit and consume events based on concrete profiles that impose vocabulary constraints as opposed to sending/receiving "general" events.

**Comment by ViktorHaag** [comment]
We simply can't make it required and put it where we propose, because that would be a breaking change.

We could be sneaky and not make it a requirement _in specification_ but make it a requirement _in certification_, but I feel dirty even suggesting that; that is, we could make it a requirement for certification that, _if you want to certify for a certain profile_ then you _must_ send events that identify _that profile_ association, otherwise, you can only certify for "General Profile".

That would skirt the objection I would usually raise when the cert does something different to the spec, and it would require everyone who wants to cert against a profile to at least demo that they could send events that way in production, and then lean on best practices to encourage them to send the property when actually sending events as well (i.e. make it "harder to do the wrong thing" which would be "behave one way in cert and a different way in production").

The objection I would usually raise is that cert-testing something not in spec is objectionable, and if something is required it should be in the spec. But, in spec in _this_ case, we could say "this property is optional, but to certify against a particular profile, you _must_ provide this property, and therefore, as a matter of best practice, you should always send this property to indicate to your event consumers what the proper interpretation profile is for your events; if you do not send it, then your event consumers must interpret the event with the rules associated with the _General profile_."

**Comment by ViktorHaag** [comment]
(To really truly make it required in specification would require a different, and much more onerous design. 8(( )
